soc/board_specific_workflows/test_general.py

```diff
@@ -0,0 +1,165 @@
+from litex.soc.integration import soc as litex_soc
+
+
+class TestGeneralSoCWorkflow(unittest.TestCase):
+    """TestCase for the GeneralSoCWorkflow class.
+    
+    All methods except 'run' are tested here.
+    """
+    def setUp(self):
+        """Sets up mocks and fakes used throughout the tests."""
+        # Generates a real argparse Namespace for testing.
+        self.target = 'test_target'
+        self.build = True
+        self.args = workflow_args.parse_workflow_args(
+            ['--target', self.target, '--build' if self.build else ''])
+
+        # Autospec the SoC.
+        self.programmer = mock.create_autospec(
+            generic_programmer.GenericProgrammer)
+        self.soc = mock.create_autospec(litex_soc.LiteXSoC, instance=True)
+        self.soc.platform = mock.create_autospec(
+            generic_platform.GenericPlatform, instance=True)
+        self.soc.platform.create_programmer.return_value = self.programmer
+        self.build_name = 'test_build_name'
+        self.soc.build_name = self.build_name
+
+        # Autospec the Builder.
+        self.builder = mock.create_autospec(builder.Builder, instance=True)
+        self.gateware_dir = 'test_gateware_dir/'
+        self.builder.gateware_dir = self.gateware_dir
+
+        self.builder_constructor = mock.create_autospec(
+            builder.Builder, return_value=self.builder)
+        # Can't easily autospec the soc_constructor -- interface inconsistent.
+        self.soc_constructor = mock.MagicMock(return_value=self.soc)
+
+    def simple_init(self):
+        """Returns a GeneralSoCWorkflow with reasonable testing parameters."""
+        return general.GeneralSoCWorkflow(self.args, self.soc_constructor,
+                                          self.builder_constructor)
+
+    def test_init(self):
+        """Tests functionality of __init__ method."""
+        flow = self.simple_init()
+
+        self.assertEqual(flow.args, self.args)
+        self.assertEqual(flow.soc_constructor, self.soc_constructor)
+        self.assertEqual(flow.builder_constructor, self.builder_constructor)
+
+    def test_make_soc(self):
+        """Tests functionality of the make_soc method."""
+        in_kwargs = {'abcd': 'efgh'}
+        soc = self.simple_init().make_soc(**in_kwargs)
+        soc_kwargs = self.soc_constructor.call_args.kwargs
+
+        self.soc_constructor.assert_called_once()
+        self.assertEqual(soc, self.soc)
+        self.assertNotIn('toolchain', soc_kwargs)
+        self.assertEqual(in_kwargs['abcd'], soc_kwargs['abcd'])
+        self.assertEqual(self.args.with_ethernet, soc_kwargs['with_ethernet'])
+        self.assertEqual(self.args.with_etherbone,
+                         soc_kwargs['with_etherbone'])
+        self.assertEqual(self.args.with_mapped_flash,
+                         soc_kwargs['with_mapped_flash'])
+
+    @mock.patch('litex.soc.integration.soc_core.soc_core_argdict')
+    def test_make_soc_argdict_call(self, mock_soc_core_argdict):
```

Comment:
These have the same issue as test_build_soc_vivado(): they're testing a function is called, but not any behavior.